packages/components/src/combobox-control/index.js

```diff
@@ -159,6 +161,16 @@ function ComboboxControl( {
 		inputContainer.current.input.focus();
 	};
 
+	// Update selections on filter change.
+	useEffect( () => {
+		if (
+			matchingSuggestions.length &&
+			matchingSuggestions.indexOf( selectedSuggestion ) < 0
+		) {
+			setSelectedSuggestion( matchingSuggestions[ 0 ] );
+		}
+	}, [ matchingSuggestions, selectedSuggestion ] );
```

Comment:
I wonder if we could extract the conditional into a well-named variable, just to help with the self-documentation of this effect? Right now it's a little obscure. Maybe something like:

```suggestion
	useEffect( () => {
		const currentSelectionNotInSuggestions = matchingSuggestions.indexOf( selectedSugestion ) < 0;
		if (
			matchingSuggestions.length &&
			currentSelectionNotInSuggestions
		) {
			// If the current selection isn't present in the list of suggestions, then automatically select the first item from the list of suggestion
			setSelectedSuggestion( matchingSuggestions[ 0 ] );
		}
	}, [ matchingSuggestions, selectedSuggestion ] );
```